### PR TITLE
New HuggingFace MolNet finetuning script

### PR DESCRIPTION
**Pull request by gabegrand**
Adds a new script to do classification/regression finetuning and eval with HuggingFace models (no simpletransformers) on MolNet tasks.

- Moves the old finetuning stuff into `old_code`
- Adds a script to compute means and stds of RDKit descriptors on a .smi file, needed for MTR pretraining.
- Adds a few more MolNet datasets to the directory

### CODE REVIEW

File: chemberta/finetune/finetune.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+import json
+import matplotlib.pyplot as plt
+import numpy as np
+import os
+import seaborn as sns
+import torch
+
+from absl import app, flags
+from scipy.special import softmax
```
Comment (walid0925) [inline chemberta/finetune/finetune.py:19]: maybe being picky but any reason we might want to use `torch` instead for softmax?

File: chemberta/finetune/finetune.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+
+def main(argv):
+    torch.manual_seed(FLAGS.seed)
+    run_dir = os.path.join(FLAGS.output_dir, FLAGS.run_name)
+    
+    tasks, (train_df, valid_df, test_df), transformers = load_molnet_dataset(FLAGS.dataset, split=FLAGS.split, df_format="chemprop")
+    assert(len(tasks) == 1)
+    
+    tokenizer = RobertaTokenizerFast.from_pretrained('seyonec/SMILES_tokenized_PubChem_shard00_160k', max_len=FLAGS.max_tokenizer_len)
```
Comment (seyonechithrananda) [inline chemberta/finetune/finetune.py]: Can you replace the path to the `seyonec/SMILES_tokenized_PubChem_shard00_160k ` with the `tokenizer_path` flag?

**Review by seyonechithrananda — APPROVED** [review]
This looks awesome! really excited to start using this. Just added one comment about adding a flag for the tokenizer path.

File: chemberta/finetune/finetune.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        model_class = RobertaForRegression
+        config.num_labels = 1
+        config.norm_mean = [np.mean(np.array(train_labels), axis=0)]
+        config.norm_std = [np.std(np.array(train_labels), axis=0)]
+
+    model = model_class.from_pretrained(FLAGS.model_dir, config=config)
+    
+    if FLAGS.freeze_base_model:
+        for name, param in model.base_model.named_parameters():
+            param.requires_grad = False
```
Comment (walid0925) [inline chemberta/finetune/finetune.py:97]: i'm 90% sure this isn't a problem - but I just wanted to make sure that we're positive that the new parameters that get added are trainable

File: chemberta/finetune/finetune.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    
+    trainer.train()
+    
+    eval_model(trainer, valid_dataset, valid_labels, FLAGS.dataset, dataset_type, os.path.join(run_dir, "results_valid"))
+    eval_model(trainer, test_dataset, test_labels, FLAGS.dataset, dataset_type, os.path.join(run_dir, "results_test"))
+
+    
+def eval_model(trainer, dataset, labels, dataset_name, dataset_type, output_dir):
+    predictions = trainer.predict(dataset)
+    fig = plt.figure(dpi=144)
```
Comment (walid0925) [inline chemberta/finetune/finetune.py:123]: i never knew you could specify `dpi` in `.figure`, always used it in `.savefig`

**Review by walid0925 — APPROVED** [review]
lgtm!

File: chemberta/finetune/finetune.py (hunk 3 above)
Comment (gabegrand) [reply chemberta/finetune/finetune.py:97]: yes - I made sure to check this. `base_model` includes only the pretrained "base" model!

File: chemberta/finetune/finetune.py (hunk 1 above)
Comment (gabegrand) [reply chemberta/finetune/finetune.py:19]: haha I tried `torch.nn.functional.softmax` and was running into some snags so I gave up

File: chemberta/finetune/finetune.py (hunk 2 above)
Comment (gabegrand) [reply chemberta/finetune/finetune.py]: done! thanks for the catch

File: chemberta/finetune/finetune.py (hunk 3 above)
Comment (schithranandanurix) [reply chemberta/finetune/finetune.py:97]: yup! this is the config structure for HF models - base_model is just the pre-trained weights
